Modules/FireForget/Get-AbortCleanKansaServersFF.ps1

```diff
@@ -0,0 +1,184 @@
+# This module is used to abort the deployment of a Kansa module from distributed-kansa servers
+# It can also be used in the event that a kansa server stalls its deployment due to too many connection
+# errors or timeouts.  The parameters can be configured to clean up result folders and error logs
+# and optionally send any uncollected results/errors to ELK for the previous n kansa jobs.
+# The default settings are set for safe operation and will only send results/errors from the last 1 job
+
+if(!(Get-Variable -Name KillRunningKansaJob -ErrorAction SilentlyContinue)){$KillRunningKansaJob = $False}
+if(!(Get-Variable -Name CleanupServer -ErrorAction SilentlyContinue)){$CleanupServer = $False}
+if(!(Get-Variable -Name SendPartialResultsToElk -ErrorAction SilentlyContinue)){$SendPartialResultsToElk = $True}
+if(!(Get-Variable -Name DeletePartialResults -ErrorAction SilentlyContinue)){$DeletePartialResults = $False}
+if(!(Get-Variable -Name NumRecentJobs -ErrorAction SilentlyContinue)){$NumRecentJobs=1}
+if(!(Get-Variable -Name KansaRemotePath -ErrorAction SilentlyContinue)){$KansaRemotePath="F:\Kansa"}
+if(!(Get-Variable -Name KansaJobPath -ErrorAction SilentlyContinue)){$KansaJobPath = ""}
+
+$thisProcess = Get-WmiObject -query 'Select * from win32_process' | where ProcessId -eq $Pid | Select Name,ProcessId,ParentProcessId
+
+$allPaths = @()
+if($KansaJobPath -eq ""){
+    $KansaJobPath = $KansaRemotePath
+    if ($NumRecentJobs -eq 0) {
+        $allPaths = gci -Path $KansaJobPath -Directory -Recurse | where Name -Match "Output_" | Sort-Object -Descending
+    } 
+    else {
+        $allPaths = gci -Path $KansaJobPath -Directory -Recurse | where Name -Match "Output_" | Sort-Object -Descending | Select -First $NumRecentJobs
+    }
+}
+else {
+    $allPaths = $KansaJobPath
+}
+
+if ($KillRunningKansaJob) {
+    $thePIDSkilled = New-Object -TypeName System.Collections.ArrayList
+    gps | where ProcessName -eq "powershell" | ForEach-object {
+        #$_.Id        
+        if (!$($_.Id -eq $thisProcess.ProcessId)) {
+            kill -Force $_.Id
+            $thePIDSkilled.add([String]$_.Id + '(powershell) ')
+        }
+    }
+
+    gps | where ProcessName -eq "wsmprovhost" | ForEach-object {
+        kill -Force $_.Id
```

Comment:
Does this kill any wsmprovhost process, even if unrelated to Kansa jobs?

Reply:
Yes, see related response for the killing of powershell.exe processes on a distributed kansa server above.  We can add a feature to restrict killed processes by user-context.  We would just need to add the -IncludeUserName switch and then use a filter "Where Name -match $username" and pass the username variable in.  Not hard, just means submitting another PR I suppose